src/plugins/data/public/search/search_interceptor.ts

```diff
@@ -171,16 +171,17 @@ export class SearchInterceptor {
       ...(abortSignal ? [abortSignal] : []),
     ];
 
-    const combinedSignal = getCombinedSignal(signals);
+    const combinedController = getCombinedController(signals);
     const cleanup = () => {
       subscription.unsubscribe();
+      combinedController.signal.removeEventListener('abort', cleanup);
+      combinedController.abort();
```

Comment:
Is the reason we need to `abort` here because we need to trigger the internal `cleanup` from `getCombinedController`? If so, maybe we can do similarly to what I've suggested above (instead of returning just the `signal` from `getCombinedSignal`, return something like `{signal, cleanup}`). Thoughts?

Reply:
> Is the reason we need to abort here because we need to trigger the internal cleanup from getCombinedController

Yes, exactly. 

> If so, maybe we can do similarly to what I've suggested above (instead of returning just the signal from getCombinedSignal, return something like {signal, cleanup})

Agree. Done!